### PR TITLE
Adding update functions for DNS Records and Reformatting

### PR DESCRIPTION
**Pull request by saiprasannasastry**
Tested all the functions maually.
Addressing #82 
Some of the parts were added for terraform bugs which internal QA has raised. Will add the remaining UT's in a different PR as this checkin is a little Urgent

### CODE REVIEW

File: object_manager.go [hunk 1]
```diff
@@ -26,20 +26,31 @@ type IBObjectManager interface {
 	GetEADefinition(name string) (*EADefinition, error)
 	CreateEADefinition(eadef EADefinition) (*EADefinition, error)
 	UpdateNetworkViewEA(ref string, addEA EA, removeEA EA) error
-	CreateHostRecord(enabledns bool, recordName string, netview string, dnsview string, cidr string, ipAddr string, macAddress string, vmID string, vmName string) (*HostRecord, error)
+}
+
+type IBOjectManager1 interface{
```
Comment (chinmayb) [inline object_manager.go]: why this change has been introduced ?

Comment (saiprasannasastry) [reply object_manager.go]: the interface was becoming big hence added these changes

File: object_manager.go [hunk 2]
```diff
@@ -427,16 +436,13 @@ func (objMgr *ObjectManager) CreateEADefinition(eadef EADefinition) (*EADefiniti
 	return newEadef, err
 }
 
-func (objMgr *ObjectManager) CreateHostRecord(enabledns bool, recordName string, netview string, dnsview string, cidr string, ipAddr string, macAddress string, vmID string, vmName string) (*HostRecord, error) {
+func (objMgr *ObjectManager) CreateHostRecord(enabledns bool, recordName string, netview string, dnsview string, cidr string, ipAddr string, macAddress string, ea EA) (*HostRecord, error) {
```
[the remark this answers is not in the document]
Comment (saiprasannasastry) [reply object_manager.go]: this was created for terraform , and it is not officially release by infoblox , hence can go to terrform and change them

File: object_manager.go (hunk 1 above)
Comment (chinmayb) [reply object_manager.go]: please take it as separate PR. we can create a interface for each object type. And group it in the objectManager interface

Comment (saiprasannasastry) [reply object_manager.go]: Sure then for now will add everything under one object?

File: object_manager.go [hunk 3]
```diff
@@ -263,25 +275,22 @@ func GetIPAddressFromRef(ref string) string {
...
-	ea := objMgr.getBasicEA(true)
-	if vmID != "" {
-		ea["VM ID"] = vmID
-	}
+	basic_ea := objMgr.getBasicEA(true)
 
-	if vmName != "" {
-		ea["VM Name"] = vmName
-	}
+	for k,v :=range ea{
```
[the remark this answers is not in the document]
Comment (saiprasannasastry) [reply object_manager.go]: Will do this

File: object_manager.go [hunk 4]
```diff
@@ -567,24 +617,64 @@ func (objMgr *ObjectManager) GetCNAMERecordByRef(ref string) (*RecordCNAME, erro
...
+	if name ==""{
+		recordCNAME.Canonical = canonical
+	}
+	if canonical ==""{
+		recordCNAME.Name = name
+	}
+	if name !="" &&canonical !=""{
+		recordCNAME.Canonical = canonical
+		recordCNAME.Name = name
+	}
```
[the remark this answers is not in the document]
Comment (saiprasannasastry) [reply object_manager.go]: it is not required to pass alias and canonical, if the user just wants to update the canonical, he can pass that and ref, if he wants to update only the alias he can pass only alias, if he wants to change the both alias and canonical name at the same time, he would pass both.

File: object_manager.go (hunk 1 above)
Comment (saiprasannasastry) [reply object_manager.go]: @chinmayb  should i add the functions back under the first inteface itself and then raise a separate PR to differentiate them?

Comment (chinmayb) [reply object_manager.go]: yes, you can do that

File: object_manager.go [hunk 5]
```diff
@@ -544,17 +547,64 @@ func (objMgr *ObjectManager) GetARecordByRef(ref string) (*RecordA, error) {
 	err := objMgr.connector.GetObject(recordA, ref, &recordA)
 	return recordA, err
 }
+func (objMgr *ObjectManager) GetARecord(recordName string,  ipAddr string) ([]RecordA,error){
```
Comment (chinmayb) [inline object_manager.go]: If this is only getting A records by `ipaddr` , rename the method that way. `GetARecord` in general should have only `mandatory` fields

Comment (chinmayb) [reply object_manager.go]: better to use `opts`
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

**Comment by saiprasannasastry** [comment]
@chinmayb abandoning this PR . will remove the get and Update functions or enhancments and raise one PR for EA changes and one more for enhancment changes
